src/main/java/com/games/online/shop/service/dao/GameShopDao.java

```
@@ -46,4 +64,43 @@ public void deleteMechanics(Mechanics mechanics) {
             log.error("Error deleting mechanics mechanicsId={}", mechanics.getId(), ex);
         }
     }
+
+    public void addOrderItem(Long orderId, OrderItem orderItem) {
+        try {
+            Optional<Order> order = orderRepository.findById(orderId);
+            if (order.isPresent()) {
+                setOrderItemId(orderId, orderItem);
+                orderItem.setOrder(order.get());
+                order.get().addOrderItem(orderItem);
+                orderRepository.save(order.get());
+                orderItemRepository.save(orderItem);
+            }
+        } catch (Exception ex) {
+            log.error("Error adding order items to order={}", orderId, ex);
+        }
+    }
+
+    public void deleteOrderItem(Long orderId, OrderItem orderItem) {
+        try {
+            Optional<Order> order = orderRepository.findById(orderId);
+            if (order.isPresent()) {
+                order.get().getOrderItems().remove(orderItem);
+                orderRepository.save(order.get());
+                orderItemRepository.delete(orderItem);
+            }
+        } catch (Exception ex) {
+            log.error("Error removing order item for order={}", orderId, ex);
+        }
+    }
+
+    private void setOrderItemId(Long orderId, OrderItem orderItem) {
+        try {
+            if (orderItem.getGame() != null) {
+                OrderItemId orderItemId = new OrderItemId(orderId, orderItem.getGame().getId());
+                orderItem.setId(orderItemId);
+            }
```

Should we throw an exception if the game is null?